### PR TITLE
refact: apply clock interface

### PR DESCRIPTION
**Pull request by HurSungYun**
use clock interface instead of time package directly for better testability

### CODE REVIEW

**Comment by HurSungYun** [comment]
Unfortunately, clock module requires Go 1.15 or above 😢

**Comment by HurSungYun** [comment]
please close this PR if you believe we should not bump Go version for this module.

**Comment by sang-w0o** [comment]
> please close this PR if you believe we should not bump Go version for this module.

[This survey](https://go.dev/blog/survey2020-results#TOC_7.) indicates that almost 76% of golang users tend to keep their golang versions up-to-date in production codes. I think it will be okay to bump it to 1.15 :)

**Comment by HurSungYun** [comment]
@sang-w0o that's great! I reverted lastest commit that implementing custom clock interface.

It would be great if you bump Go version of this project. After that, I'll update this branch

File: minimemcached.go [hunk 1]
```diff
@@ -50,19 +51,33 @@ type Item struct {
...
 	}
+
 	return &m
 }
 
+func WithClock(clk clock.Clock) MiniMemcachedOption {
+	return func(m *MiniMemcached) {
+		m.clock = clk
+	}
+}
```
Comment (sang-w0o) [inline minimemcached.go:76]: It would be nice to have some comment added for this function!

Comment (HurSungYun) [reply minimemcached.go:76]: comments are added here: https://github.com/daangn/minimemcached/pull/2/commits/e2def61c602bfb880dc2ff4f316b62b959b6fedd

**Review by sang-w0o — APPROVED** [review]
Looks great! 💯

**Review by novemberde — APPROVED** [review]
LGTM 💯 
Thank you for the contribution!

**Pull request merged**